vcr/pe/presentation_submission.go

```diff
@@ -137,3 +140,45 @@ func (b *PresentationSubmissionBuilder) Build(format string) (PresentationSubmis
 
 	return presentationSubmission, signInstructions, nil
 }
+
+// Validate validates the Presentation Submission to the Verifiable Presentation and Presentation Definition and returns the mapped credentials.
+// The credentials will be returned as map with the InputDescriptor.Id as key.
+// It assumes credentials of the presentation only map in 1 way to the input descriptors.
+func (s PresentationSubmission) Validate(presentation vc.VerifiablePresentation, definition PresentationDefinition) (map[string]vc.VerifiableCredential, error) {
```

Comment:
`Match` should support multiple VPs first, since it returns the input descriptor mappings, which must refer to the correct VP. Thus the change is mostly in the `Match` & friends. I'd prefer to keep it out of scope for this change.

Reply:
Then don't use match but the builder?